### PR TITLE
Ignore hidden directories when returning the finaldir

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

File: easybuild/tools/filetools.py [hunk 1]
```diff
@@ -472,12 +472,11 @@ def find_base_dir():
...
     def get_local_dirs_purged():
         # e.g. always purge the log directory
+        # and hidden directories
         ignoreDirs = ["easybuild"]
 
         lst = os.listdir(os.getcwd())
-        for ignDir in ignoreDirs:
-            if ignDir in lst:
-                lst.remove(ignDir)
+        lst = [Dir for Dir in lst if not Dir.startswith('.') or Dir in ignoreDirs]
```
Comment (boegel) [inline easybuild/tools/filetools.py]: don't use `Dir`, and `dir` is a keyword, so just use `d` (single-letter variables are fine in a list comprehension)

File: easybuild/tools/filetools.py [hunk 2]
```diff
@@ -472,12 +472,11 @@ def find_base_dir():
     """
     def get_local_dirs_purged():
         # e.g. always purge the log directory
+        # and hidden directories
         ignoreDirs = ["easybuild"]
```
Comment (boegel) [inline easybuild/tools/filetools.py]: kill camel case while you're at it: make this `ignore_dirs`

**Comment by boegel** [comment]
@wpoely86: let's enhance the unit tests too, maybe add a new test in test/framework/filetools.py dedicated to `find_base_dir`?

**Comment by wpoely86** [comment]
@boegel please recheck

**Comment by boegel** [comment]
Looks great now, I'm happy to see that writing a test made you catch a bug in your one-line change. ;-)

Going in, thanks @wpoely86!

**Pull request merged**